src/main.rs

```diff
@@ -187,14 +213,130 @@ where command is one of:
 
                 let mut db = db_conn.begin_save_point();
                 let mut ast = parse(&content).expect("Failed to parse program");
-                type_check(&"transient", &mut ast, &mut db, false)
-                    .unwrap_or_else(|e| {
-                        eprintln!("Type check error.\n{}", e);
-                        process::exit(1);
-                    });
+                let mut contract_analysis = type_check(&"transient", &mut ast, &mut db, false).unwrap_or_else(|e| {
+                    eprintln!("Type check error.\n{}", e);
+                    process::exit(1);
+                });
 
+                match argv.last() {
+                    Some(s) if s == "--output_analysis" => {
+                        println!("{}", contract_analysis.serialize());
+                    },
+                    _ => {}
+                }
+                
                 return
             },
+            "repl" => {
+                let mut db_conn = match ContractDatabaseConnection::memory() {
+                    Ok(db) => db,
+                    Err(error) => {
+                        eprintln!("Could not open vm-state: \n{}", error);
+                        process::exit(1);
+                    }
+                };
+
+                let mut outer_sp = db_conn.begin_save_point_raw();                    
+                let mut db = ContractDatabase::from_savepoint(outer_sp);
+
+                let mut vm_env = OwnedEnvironment::new(&mut db);
+                let mut exec_env = vm_env.get_exec_environment(None);
+
+                let mut analysis_db_conn = AnalysisDatabaseConnection::memory();
+
+                let mut reader = match linefeed::Interface::new("local-repl") {
```

Comment:
We don't need the `linefeed` library if all we're going to be doing is reading strings from stdin and storing a history.  Also, what about multi-line contracts?